sql/src/main/java/io/crate/analyze/TablePropertiesAnalyzer.java

```diff
@@ -140,243 +49,54 @@ public static void analyze(TableParameter tableParameter,
-            }
-        }
-        if (!properties.isEmpty()) {
-            Map<String, Expression> tableProperties = properties.properties();
-            validateTableProperties(tableParameterInfo, tableProperties.keySet());
-
-            for (String setting : tableParameterInfo.supportedSettings()) {
-                String settingName = ES_TO_CRATE_SETTINGS_MAP.get(setting);
-                if (tableProperties.containsKey(settingName)) {
-                    SettingsApplier settingsApplier = SETTINGS_APPLIER.get(setting);
-                    settingsApplier.apply(tableParameter.settingsBuilder(), parameters, tableProperties.get(settingName));
-                }
-            }
-            for (String mappingEntry : tableParameterInfo.supportedMappings()) {
-                String mappingName = ES_TO_CRATE_MAPPINGS_MAP.get(mappingEntry);
-                if (tableProperties.containsKey(mappingName)) {
-                    MappingsApplier mappingsApplier = MAPPINGS_APPLIER.get(mappingEntry);
-                    mappingsApplier.apply(tableParameter.mappings(), parameters, tableProperties.get(mappingName));
-                }
-            }
-        }
-
+        Map<String, Setting> settingMap = tableParameterInfo.supportedSettings();
+        Map<String, Setting> mappingsMap = tableParameterInfo.supportedMappings();
+
+        GenericPropertiesConverter.settingsFromProperties(
+            tableParameter.settingsBuilder(),
+            properties,
+            parameters,
+            settingMap,
+            withDefaults,
+            mappingsMap::containsKey,
+            INVALID_MESSAGE);
+
+        GenericPropertiesConverter.settingsFromProperties(
+            tableParameter.mappingsBuilder(),
+            properties,
+            parameters,
+            mappingsMap,
+            withDefaults,
+            settingMap::containsKey,
+            INVALID_MESSAGE);
     }
 
     public static void analyze(TableParameter tableParameter,
                                TableParameterInfo tableParameterInfo,
                                List<String> properties) {
-        validateTableProperties(tableParameterInfo, properties);
-
-        for (String setting : tableParameterInfo.supportedSettings()) {
-            String settingName = ES_TO_CRATE_SETTINGS_MAP.get(setting);
-            if (properties.contains(settingName)) {
-                SettingsApplier settingsApplier = SETTINGS_APPLIER.get(setting);
-                tableParameter.settingsBuilder().put(settingsApplier.getDefault());
+        Map<String, Setting> settingMap = tableParameterInfo.supportedSettings();
+        Map<String, Setting> mappingsMap = tableParameterInfo.supportedMappings();
+
+        for (String name : properties) {
+            Setting setting = settingMap.get(name);
+            boolean isMappingEntry = false;
+            if (setting == null) {
+                setting = mappingsMap.get(name);
+                isMappingEntry = true;
```

Comment:
From reading just the diff it is a bit incomprehensible what's being done here. It would help if the high level action had a better name, or if we had some comments here.